File: nodes/string-trim/string-trim.process.py
import re

async def process(inputs, settings, config, nodeConfig):
  text = str(inputs.get("text", ""))
  chars = inputs.get("chars")
  mode = settings.get("mode", "both")
  
  if chars:
    pattern = ""
    if mode in ["both", "start"]:
      pattern += f"^[{re.escape(chars)}]+"
    if mode in ["both", "end"]:
      if pattern:
        pattern += "|"
      pattern += f"[{re.escape(chars)}]+$"
    return {
      "text": re.sub(pattern, "", text)
    }
  
  if mode == "start":
    return {"text": text.lstrip()}
  elif mode == "end":
    return {"text": text.rstrip()}
  else:
    return {"text": text.strip()} 

File: sdks/nodejs/nodes/string-trim/string-trim.process.js
export default async ({inputs, settings, config}) => {

  const text = String(inputs.text || "");
  const chars = inputs.chars;
  const mode = settings.mode || "both";

  // If specific characters are provided, use them for trimming
  if (chars) {
    const regex = new RegExp(`^[${chars}]+|[${chars}]+$`, mode === "start" ? "^" : mode === "end" ? "$" : "g");
    return { text: text.replace(regex, "") };
  }

  // Otherwise use standard whitespace trimming
  switch (mode) {
    case "start":
      return { text: text.trimStart() };
    case "end":
      return { text: text.trimEnd() };
    default:
      return { text: text.trim() };
  }
}; 